### PR TITLE
[DB-13027] Implement reporting of XML functions as Unsupported Query Construct in migration assessment

### PR DESCRIPTION
**Pull request by sanyamsinghal**
- covers simple SELECT stmts with xml functions, in target_list, from clause or where clause of the statement

### CODE REVIEW

File: yb-voyager/src/queryparser/unsupportedConstructs.go [hunk 1]
```diff
@@ -315,8 +328,115 @@ func containsSystemColumnsInWhereClause(whereClause *pg_query.Node) bool {
...
+		if resTarget == nil {
+			return false
+		}
+
+		xmlExpr := resTarget.Val.GetXmlExpr()
+		if xmlExpr != nil {
+			return true
+		}
+	}
+	return false
```
Comment (priyanshi-yb) [inline yb-voyager/src/queryparser/unsupportedConstructs.go:370]: we can also have xml functions in the target list e.g. `SELECT id, xpath('/person/name/text()', data) AS name from xml_example;` ?

File: yb-voyager/src/queryparser/unsupportedConstructs.go [hunk 2]
```diff
@@ -315,8 +328,115 @@ func containsSystemColumnsInWhereClause(whereClause *pg_query.Node) bool {
...
+	}
+
+	if funcCallNode := whereClause.GetFuncCall(); funcCallNode != nil {
+		funcList := funcCallNode.Funcname
+		functionName := funcList[len(funcList)-1].GetString_().Sval
+		if slices.Contains(xmlFunctions, functionName) {
+			return true
+		}
+	}
+
```
Comment (priyanshi-yb) [inline yb-voyager/src/queryparser/unsupportedConstructs.go:439]: Thinking loud, can there be any case in which where clause can have the `XmlExpr` way as in the above target and from clause?

Comment (sanyamsinghal) [reply yb-voyager/src/queryparser/unsupportedConstructs.go:439]: yes @priyanshi-yb , one case can be - WHERE clause having a subquery

Comment (sanyamsinghal) [reply yb-voyager/src/queryparser/unsupportedConstructs.go:439]: But let me cover those later, need to think it through.

File: yb-voyager/src/queryparser/unsupportedConstructs.go (hunk 1 above)
Comment (sanyamsinghal) [reply yb-voyager/src/queryparser/unsupportedConstructs.go:378]: done

File: yb-voyager/src/queryparser/unsupportedConstructs.go (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/src/queryparser/unsupportedConstructs.go:447]: okay

**Review by priyanshi-yb — APPROVED** [review]
LGTM

**Pull request merged**